### PR TITLE
Move seft upload

### PR DESCRIPTION
**Pull request by insacuri**
# What and why?

Previously we would get the file from the user and make a request to collection-instrument to actually act on it.  Now we'll just act on it ourselves by putting the file in the bucket and the message into pubsub.

The code around this was.... not great, so as well as copying the code across from collection-instrument, it's been refactored so it should be more efficient and easier to maintain.

As well as this, I've moved a bunch of tests that don't rely on any external sources from integration to unit.  The theory was that integration tests (that need and/or test real backing services) would live in their own section, but there are a lot of tests in the integration folder that don't rely on anything external (because of mocking, etc)

# How to test?

- Deploy this to your namespace
- Run SEFT acceptance tests (easily done through spinnaker)
- Log in as user and try and upload a SEFT response.
- Check in the bucket in your namespace that it's appeared as expected and that a pubsub message was put onto the queue.

You'll notice there is a lot of spammy logging, it's because we run the app at DEBUG level in development, but we run it at INFO level further downstream in preprod and prod so that'll go away.  You can verify this yourself by setting the LOGGING_LEVEL in your environment.

# Trello

### CODE REVIEW

**Review by anwilkie — COMMENTED** [review]
The code looked okay to me, but when I:

1. Logged into response-ops, created new collection exercise, filled in dates, added sample and collection instrument, and set exercise to live.
2. Fetched enrolment code for respondent.
3. Logged in as respondent on frontstage and entered enrolment code.
4. Downloaded SEFT file and submitted response.

I then received an error and the SEFT response didn't end up in my envrionment's SEFT response bucket. Here is the output from the frontstage logs:

`{"filename": "49900000006F_1912_064_20220225105540.xlsx.gpg", "case_id": "49054ba3-92b5-4ea0-b2bb-f6188e4b2f5b", "survey_id": "064", "tx_id": "e61400eb-17b2-4ddd-a82f-04fda9524c88", "project": "ras-rm-sandbox", "bucket": "test-bucket", "url": "http://localhost:8082/surveys/upload-survey?case_id=49054ba3-92b5-4ea0-b2bb-f6188e4b2f5b&business_party_id=fa15ebae-f77f-45e3-b25f-5f9d85270577&survey_short_name=QIFDI", "status_code": 500, "event": "Generic exception generated", "severity": "error", "level": "error", "service": "ras-frontstage", "exception": ["Traceback (most recent call last):", "  File '/usr/local/lib/python3.10/site-packages/flask/app.py', line 1516, in full_dispatch_request", "    rv = self.dispatch_request()", "  File '/usr/local/lib/python3.10/site-packages/flask/app.py', line 1502, in dispatch_request", "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)", "  File '/app/frontstage/common/authorisation.py', line 61, in extract_session_wrapper", "    return original_function(session, *args, **kwargs)", "  File '/app/frontstage/views/surveys/upload_survey.py', line 64, in upload_survey", "    collection_instrument_controller.upload_collection_instrument(", "  File '/app/frontstage/controllers/collection_instrument_controller.py', line 111, in upload_collection_instrument", "    gcp_survey_response.upload_seft_survey_response(case, file_contents, file_name, survey_ref)", "  File '/app/frontstage/controllers/gcp_survey_response.py', line 73, in upload_seft_survey_response", "    results = self.put_file_into_gcp_bucket(file_contents, file_name)", "  File '/app/frontstage/controllers/gcp_survey_response.py', line 120, in put_file_into_gcp_bucket", "    encrypter = GNUEncrypter(gnugpg_secret_keys)", "  File '/app/frontstage/controllers/gnu_encryptor.py', line 12, in __init__", "    self.gpg.import_keys(public_key.encode('utf-8'))", "AttributeError: 'NoneType' object has no attribute 'encode'"], "created_at": "2022-02-25T10:551645786540"}`

I'm not sure if I did something wrong? I did the same steps I usually do when testing out SEFT upload. Based on what I remember when I wrote the code for encrypting the SEFT with a key, I think this error shows up when there's an issue with the key itself, or with how the code is reading in the key.

**Review by anwilkie — APPROVED** [review]

File: Pipfile [hunk 1]
```diff
@@ -26,7 +26,10 @@ requestsdefaulter = "*"
 pycryptodome = "*"
 flask-talisman = "*"
 email_validator = "*"
-google-cloud-pubsub = "==1.7.0"
+google-cloud-pubsub = "*"
+google-cloud-storage = "*"
+python-gnupg = "==0.4.7"
```
Comment (creativeappdevamit) [inline Pipfile:31]: why? this is the package which should get continuous updates. may be lets unpin it in another card.

**Review by creativeappdevamit — APPROVED** [review]

**Pull request merged**